src/party_context_menu.c

```diff
@@ -0,0 +1,1290 @@
+    PartyMenuContextButtonAnimData *unk = &partyMenu->contextMenuButtonAnim;
+
+    unk->autoAnimTimer = 0;
+    unk->buttonAnimState = 0;
+    unk->template = &cursor->menu;
+    unk->numItems = cursor->numItems;
+    unk->selection = cursor->selection;
+    unk->state = cursor->state;
+    unk->followUpState = followUpState;
+    unk->active = TRUE;
+}
+
+void PartyMenu_StartContextMenuButtonAnim(PartyMenuStruct *partyMenu, int selection, int followUpState, BOOL restartAnim) {
+    PartyMenuContextButtonAnimData *unk = &partyMenu->contextMenuButtonAnim;
+
+    unk->autoAnimTimer = 0;
+    unk->buttonAnimState = 0;
+    unk->template = NULL;
+    unk->selection = selection;
+    unk->state = Get2dSpriteCurrentAnimSeqNo(partyMenu->sprites[selection]) & 2;
+    if (restartAnim == TRUE) {
+        Sprite_SetAnimCtrlCurrentFrame(partyMenu->sprites[selection], 0);
+        Set2dSpriteAnimSeqNo(partyMenu->sprites[selection], unk->state);
+    }
+    unk->followUpState = followUpState;
+    unk->active = TRUE;
+}
+
+BOOL PartyMenu_AnimateContextMenuButtonPress(PartyMenuStruct *partyMenu) {
+    PartyMenuContextButtonAnimData *unk = &partyMenu->contextMenuButtonAnim;
+
+    switch (unk->buttonAnimState) {
+    case 0:
+        if (unk->template != NULL) {
+            sub_0207E3A8(partyMenu, unk->numItems, unk->selection, unk->state, 2);
+            PartyMenu_PrintContextMenuItemText(partyMenu, unk->template, unk->numItems, unk->selection, unk->state, FALSE);
+            BgCommitTilemapBufferToVram(partyMenu->bgConfig, GF_BG_LYR_MAIN_0);
+        } else {
+            Sprite_SetAnimCtrlCurrentFrame(partyMenu->sprites[unk->selection], 0);
+            Set2dSpriteAnimSeqNo(partyMenu->sprites[unk->selection], unk->state);
+        }
+        ++unk->buttonAnimState;
+        break;
+    case 1:
+        ++unk->autoAnimTimer;
+        if (unk->autoAnimTimer == 2) {
+            if (unk->template != NULL) {
+                sub_0207E3A8(partyMenu, unk->numItems, unk->selection, unk->state, 1);
+                PartyMenu_PrintContextMenuItemText(partyMenu, unk->template, unk->numItems, unk->selection, unk->state, TRUE);
+                BgCommitTilemapBufferToVram(partyMenu->bgConfig, GF_BG_LYR_MAIN_0);
+            } else {
+                Sprite_SetAnimCtrlCurrentFrame(partyMenu->sprites[unk->selection], 0);
+                Set2dSpriteAnimSeqNo(partyMenu->sprites[unk->selection], unk->state + 1);
+            }
+            unk->autoAnimTimer = 0;
+            ++unk->buttonAnimState;
+        }
+        break;
+    case 2:
+        ++unk->autoAnimTimer;
+        if (unk->autoAnimTimer == 2) {
+            unk->active = FALSE;
+            return FALSE;
+        }
+        break;
+    }
+
+    return TRUE;
+}
+
+static void addFiveWindows(BgConfig *bgConfig, Window *window, const WindowTemplate *template) {
```

Comment:
follow function naming conventions

Reply:
exception to the convention is local functions that don't operate on the file's primary object